src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -968,13 +976,65 @@ void RaftReplDev::cp_flush(CP*) {
         // Not dirtied since last flush ignore
         return;
     }
+
+    std::unique_lock lg{m_sb_mtx};
     m_rd_sb->compact_lsn = clsn;
-    m_rd_sb->commit_lsn = lsn;
+    m_rd_sb->durable_commit_lsn = lsn;
     m_rd_sb->checkpoint_lsn = lsn;
     m_rd_sb->last_applied_dsn = m_next_dsn.load();
     m_rd_sb.write();
     m_last_flushed_commit_lsn = lsn;
 }
 
 void RaftReplDev::cp_cleanup(CP*) {}
+
+void RaftReplDev::on_log_found(logstore_seq_num_t lsn, log_buffer buf, void* ctx) {
+    // apply the log entry if the lsn is between checkpoint lsn and durable commit lsn
+    if (m_rd_sb->checkpoint_lsn >= lsn || m_rd_sb->durable_commit_lsn <= lsn) { return; }
```

Comment:
Nit, interesting that you are checking this way than other way round. This comment is subjective, but to me its easier to read my lsn is out-of-range to whats stored, than other way round :-)

```
if (lsn < m_rd_sb->checkpoint_lsn || lsn < m_rd_sb->durable_commit_lsn) { return; }
```

Reply:
yeah, makes sense. I was following the block diagram in the doc and the conditional came out to be this :)